rskj-core/src/main/java/co/rsk/signing/PlainMessage.java

```diff
@@ -0,0 +1,44 @@
+package co.rsk.signing;
+
+import java.util.Arrays;
+
+/**
+ * Immutable plain byte array message
+ *
+ * @author Ariel Mendelzon
+ */
+public class PlainMessage extends Message {
+    private final byte[] message;
+
+    public PlainMessage(byte[] message) {
+        // Save a copy
+        this.message = copy(message);
+    }
+
+    @Override
+    public byte[] getBytes() {
+        return copy(message);
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I think this is harder than you might think: https://stackoverflow.com/a/12239416.
If we're not using `equals`, can we just remove it?

Reply:
The fact that something is not being used doesn't imply that its observational equality is not well defined and should be reflected in code. That's what the `equals` and `hashCode` do, meaning in turn that this allows for several instances of this class to live together and interact meaningfully.

On the stackoverflow link: can you please explain on here what you mean?

Reply:
Looks OK to me

Reply:
I basically think that we shouldn't implement equality if we're not using equality for any purpose (e.g. comparison, storing in a `Set`, etc.).

The SO link tried to show that implementing equality in a class hierarchy is harder than it seems, and that makes me think it's even a better idea to leave this out.

Reply:
Alright, difference in opinions regarding design. Probably deserves a broader discussion involving other team members.

WIth respect to the hierarchy: I see only one abstract and one concrete class. And equality is defined in the concrete. I don't see any problems with the hierarchy.